internal/provider/resources/flow.go

```diff
@@ -0,0 +1,372 @@
+			"Error creating flow",
+			fmt.Sprintf("Could not create flow, unexpected error: %s", err),
+		)
+
+		return
+	}
+
+	resp.Diagnostics.Append(copyFlowToModel(ctx, flow, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *FlowResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var model FlowResourceModel
+
+	// Populate the model from state and emit diagnostics on error
+	resp.Diagnostics.Append(req.State.Get(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// if model.ID.IsNull() && model.Handle.IsNull() {
+	// 	resp.Diagnostics.AddError(
+	// 		"Both ID and Handle are unset",
+	// 		"This is a bug in the Terraform provider. Please report it to the maintainers.",
+	// 	)
+
+	// 	return
+	// }
+
+	client, err := r.client.Flows(model.AccountID.ValueUUID(), model.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating deployment client",
+			fmt.Sprintf("Could not create deployment client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
+		)
+	}
+
+	// A deployment can be imported + read by either ID or Handle
+	// If both are set, we prefer the ID
+	var flow *api.Flow
+	if !model.ID.IsNull() {
+		var flowID uuid.UUID
+		flowID, err = uuid.Parse(model.ID.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddAttributeError(
+				path.Root("id"),
+				"Error parsing Flow ID",
+				fmt.Sprintf("Could not parse Flow ID to UUID, unexpected error: %s", err.Error()),
+			)
+
+			return
+		}
+
+		flow, err = client.Get(ctx, flowID)
+	}
+	// } else if !model.Handle.IsNull() {
+	// 	var deployments []*api.Deployment
+	// 	deployments, err = client.List(ctx, []string{model.Handle.ValueString()})
+
+	// 	// The error from the API call should take precedence
+	// 	// followed by this custom error if a specific deployment is not returned
+	// 	if err == nil && len(deployments) != 1 {
+	// 		err = fmt.Errorf("a deployment with the handle=%s could not be found", model.Handle.ValueString())
+	// 	}
+
+	// 	if len(deployments) == 1 {
+	// 		deployment = deployments[0]
+	// 	}
+	// }
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error refreshing flow state",
+			fmt.Sprintf("Could not read Flow, unexpected error: %s", err.Error()),
+		)
+
+		return
+	}
+
+	resp.Diagnostics.Append(copyFlowToModel(ctx, flow, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *FlowResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Unsupported
```

Comment:
Any reason this is marked as unsupported? https://app.prefect.cloud/api/docs#tag/Flows/operation/update_flow_api_accounts__account_id__workspaces__workspace_id__flows__id__patch

Reply:
The default behaviour would be delete and recreate on a change. 
It's unclear if the use case to update in place adds much value.
Keen to understand if there's requirement.

Reply:
My suggestion would be to see if customers need this capability. 
From what i can see the effort to update is not adding much.

Reply:
@parkedwards - could you weigh in here since you have more context? I'm thinking that if we choose not to support `Update` then let's remove the method entirely to avoid confusion while developing.

Reply:
i think @skyscrapr's intuition is correct here. it looks like we only accept changes to the `tags` parameter on the flow resource. practically, we'd probably re-create the flow on the API if we _**did**_ need to make any material change

that said, i do think we need to keep `Update` in place, as we need to implement the `resource.Resource` interface

Reply:
Yes. The method has to remain. 
But the standard way to not support update is to leave this method blank. 

The core will then treat any changes in the config as a destroy/create for the resource.

Reply:
Got it, thank you both 👍🏼

Reply:
@skyscrapr - given the result of the discussion here, it sounds to me like we don't want users to be able to update Flows and instead force a replace. We can use PlanModifiers for that. For example:

```diff
diff --git a/internal/provider/resources/flow.go b/internal/provider/resources/flow.go
index d568c2b..669bae3 100644
--- a/internal/provider/resources/flow.go
+++ b/internal/provider/resources/flow.go
@@ -12,6 +12,7 @@ import (
        "github.com/hashicorp/terraform-plugin-framework/resource"
        "github.com/hashicorp/terraform-plugin-framework/resource/schema"
        "github.com/hashicorp/terraform-plugin-framework/resource/schema/listdefault"
+       "github.com/hashicorp/terraform-plugin-framework/resource/schema/listplanmodifier"
        "github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
        "github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
        "github.com/hashicorp/terraform-plugin-framework/types"
@@ -127,6 +128,9 @@ func (r *FlowResource) Schema(_ context.Context, _ resource.SchemaRequest, resp
                                Optional:    true,
                                Computed:    true,
                                Default:     listdefault.StaticValue(defaultEmptyTagList),
+                               PlanModifiers: []planmodifier.List{
+                                       listplanmodifier.RequiresReplace(),
+                               },
                        },
                },
        }
```

Without this change, when I try to update an existing Flow resource with an additional tag, I get:

```
Terraform will perform the following actions:

  # prefect_flow.flow will be updated in-place
  ~ resource "prefect_flow" "flow" {
      ~ created = "2024-06-13T19:29:06Z" -> (known after apply)
        id      = "54e09681-9535-4f0b-9a15-ffd999c7187e"
        name    = "test-flow"
      ~ tags    = [
            "test",
          + "test2",
        ]
      ~ updated = "2024-06-13T19:29:06Z" -> (known after apply)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

prefect_flow.flow: Modifying... [id=54e09681-9535-4f0b-9a15-ffd999c7187e]
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to prefect_flow.flow, provider "provider[\"registry.terraform.io/prefecthq/prefect\"]" produced an unexpected new value:
│ .tags: element 1 has vanished.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

_With_ the diff applied using the PlanModifier, I now get something more helpful:

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # prefect_flow.flow must be replaced
-/+ resource "prefect_flow" "flow" {
      ~ created = "2024-06-13T19:29:06Z" -> (known after apply)
      ~ id      = "54e09681-9535-4f0b-9a15-ffd999c7187e" -> (known after apply)
        name    = "test-flow"
      ~ tags    = [ # forces replacement
            "test",
          + "test2",
        ]
      ~ updated = "2024-06-13T19:29:06Z" -> (known after apply)
    }

Plan: 1 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

prefect_flow.flow: Destroying... [id=54e09681-9535-4f0b-9a15-ffd999c7187e]
prefect_flow.flow: Destruction complete after 0s
prefect_flow.flow: Creating...
prefect_flow.flow: Creation complete after 0s [id=c153db34-3aa9-4551-b07b-7e58df3ac9e6]

Apply complete! Resources: 1 added, 0 changed, 1 destroyed
```

What do you think?

Reply:
Looks good to me. I added a test to check for updating tags in the flow resource. I had to add the plan modifier suggested to get the correct functionality. tests pass so confirms that changing tags will work correctly.

good suggestion thanks!